src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class DecisionDagBuilder
+    {
+        private void MakeTestsAndBindingsForLengthAndListPatterns(BoundDagTemp input, BoundRecursivePattern recursive, ArrayBuilder<BoundPatternBinding> bindings, ArrayBuilder<Tests> tests)
+        {
+            if (recursive.HasErrors)
+                return;
+
+            Debug.Assert(recursive.LengthProperty is not null || input.Type.IsSZArray());
+            var syntax = recursive.Syntax;
+            var lengthProperty = recursive.LengthProperty ?? (PropertySymbol)_compilation.GetSpecialTypeMember(SpecialMember.System_Array__Length);
+            var lengthEvaluation = new BoundDagPropertyEvaluation(syntax, lengthProperty, input);
+            tests.Add(new Tests.One(lengthEvaluation));
+            var lengthTemp = new BoundDagTemp(syntax, _compilation.GetSpecialType(SpecialType.System_Int32), lengthEvaluation);
+
+            if (recursive.LengthPattern is not null)
+            {
+                tests.Add(MakeTestsAndBindings(lengthTemp, recursive.LengthPattern, bindings));
+            }
+
+            if (recursive.ListPatternClause is { Subpatterns: var subpatterns } clause)
+            {
+                Debug.Assert(!subpatterns.IsDefaultOrEmpty);
+                Debug.Assert(subpatterns.Count(p => p.Kind == BoundKind.SlicePattern) <= 1);
+
+                if (clause.HasSlice &&
+                    subpatterns.Length == 1 &&
+                    subpatterns[0] is BoundSlicePattern { Pattern: null })
```

Comment:
What about a slice with no subpattern in the middle or tail of a list pattern? Do we need to skip emitting a test for those too?
Example: `{ 1, .., 2 }`, `{ 1, .. }`

Reply:
That does not require a Slice and therefore nothing to emit.
As for the length, it's already handled below as usual.